kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::string FromJSONCheckResultProcess::GetNodeIdentifier(NodeType& rNode)
+{
+    if (this->Is(USE_NODE_COORDINATES)) {
+        const SizeType digits = 6;
+        std::stringstream ss;
+        ss.setf( std::ios::fixed );
+        ss << "X_" << std::setprecision( digits ) << rNode.X0() << "_Y_" << std::setprecision( digits ) << rNode.Y0() << "_Z_" << std::setprecision( digits ) << rNode.Z0();
+        return ss.str();
+    } else {
+        return std::to_string(rNode.Id());
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::NodesArrayType& FromJSONCheckResultProcess::GetNodes(const Flags* pFlag)
+{
+    KRATOS_TRY;
+
+    if (this->IsNot(NODES_CONTAINER_INITIALIZED)) {
+        auto& r_nodes_array = this->Is(CHECK_ONLY_LOCAL_ENTITIES) ? mrModelPart.GetCommunicator().LocalMesh().Nodes() : mrModelPart.Nodes();
+        const auto it_node_begin = r_nodes_array.begin();
+        const int number_of_nodes = static_cast<int>(r_nodes_array.size());
+
+        #pragma omp parallel firstprivate(number_of_nodes)
+        {
+            NodesArrayType nodes_buffer;
+
+            # pragma omp for schedule(guided, 512) nowait
+            for (int i = 0; i < number_of_nodes; ++i) {
+                auto it_node = it_node_begin + i;
+                if (CheckFlag(*it_node, pFlag)) nodes_buffer.push_back(*(it_node.base()));
+
+            }
+
+            // We merge all the sets in one thread
+            #pragma omp critical
+            {
+                for (auto it_node = nodes_buffer.begin(); it_node < nodes_buffer.end(); ++it_node)
+                    mNodesArray.push_back(*(it_node.base()));
+            }
+        }
+        // Set flag
+        this->Set(NODES_CONTAINER_INITIALIZED, true);
+    }
+
+    return mNodesArray;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::ElementsArrayType& FromJSONCheckResultProcess::GetElements(const Flags* pFlag)
+{
+    KRATOS_TRY;
+
+    if (this->IsNot(ELEMENTS_CONTAINER_INITIALIZED)) {
+        auto& r_elements_array = this->Is(CHECK_ONLY_LOCAL_ENTITIES) ? mrModelPart.GetCommunicator().LocalMesh().Elements() : mrModelPart.Elements();
```

Comment:
The elements are always local (there are no ghost elements).

Reply:
I mean, you can directly check the elements in the local mesh.

Reply:
Idem

Reply:
Idem

Reply:
Idem